src/lucky/errors.cr

```diff
@@ -7,8 +7,10 @@ module Lucky
 
   # Raised when a route could not be found
   class RouteNotFoundError < Error
-    def initialize(method : String, path : String)
-      super "Could not find route matching #{method} #{path}"
+    property context : HTTP::Server::Context
```

Comment:
This can be a `getter` since it shouldn't be set outside this class. It also doesn't need a type since you defined it in the initialize. So in the end it could be: `getter context, params`